### ISSUE TITLE
Non-latin Command Names cause Message Content to be cut off

### ISSUE THREAD

**Issue by lem0nify** [serenity-rs/serenity#442]
args.full() works incorrectly (cuts off some first symbols of args line) if used with non-latin command.
This makes me use constructions like
```Rust
let argline = message.content.split(" ").skip(1).collect::<Vec<&str>>().join(" ");
```
instead of just
```Rust
let argline = args.full().to_string();
```

**Comment by acdenisSK** [comment]
Could you provide some example input where this problem occurs?
Because what `Args::full` does is return the command's "content" that was parsed from the full message, i. e:
```rust
let msg = "?echo hello world";
let content = parse_content(&msg);
assert_eq!(content, "hello world");
```
And if this problem seems to be reproducible in all (/most) cases, then there's a parsing bug in the framework itself.

**Comment by lem0nify** [comment]
@acdenisSK As I said, the problem occurs every time when args.full() used in the body of command which contains non-latin symbols, e.g. cyrillic, like in example below:
```Rust
pub fn generate_framework() -> StandardFramework {
    StandardFramework::new()
        .configure(|c| c.prefix("!"))
        .cmd("echo", echo)
        .cmd("эхо", echo)
}

command!(echo(_context, message, args) {
    message.reply(args.full());
});
```
![example](https://user-images.githubusercontent.com/17163615/48972770-fdb7f780-f042-11e8-9b4e-34768c9a17bc.png)

if command is longer, to longer part of arg line will be cut off
![example](https://user-images.githubusercontent.com/17163615/48972804-82a31100-f043-11e8-8acf-e09be649be66.png)

**Comment by acdenisSK** [comment]
Ah. You weren't clear enough when you said `if used with non-latin command`. I thought it was something after the command invocation, not the command's *name* itself.

This, however, is its own separate problem. We never had any rationale for supporting unicode in commands' names when designing the framework, and probably never will. You can consider yourself lucky that this even works and that your bot hadn't panic yet when doing this.

Anyhow, I'm sorry, but I'll have to close this as It's unlikely we'll be fixing this anytime soon.
(Though we may bring it up in the future, maybe.)

**Comment by acdenisSK** [comment]
I know. But special characters in a command's name can make it a big turnoff for people wishing to use it, as it's far more likely that the average person has a keyboard containing the entire latin alphabet, plus punctuation marks, rather than the cyrilic alphabet (or that they even have the cyrilic layout on-hand for quick use).

Furthermore, it'd complicate parsing in the framework for specific cases that have few chances to occur. That would be uncomfortable to support all the time (unicode is already a bitch as it is, too).

So, even though I do understand your motive, you'll have to live with using latin for your commands.

**Comment by Lakelezz** [comment]
@lem0nify, we convinced Denis to support it : D

**Comment by Lakelezz** [comment]
Thanks for opening this issue and sorry for the initial inconvenience.
This issue is resolved on the `v0.6.x`-branch! : ) We are planning on releasing an RC.
